JavaScript/TypeScript.sublime-syntax

```diff
@@ -1047,11 +1047,15 @@ contexts:
       pop: 1
     - match: \.\.\.
       scope: keyword.operator.spread.js
-    - match: '{{identifier_name}}'
-      scope: variable.parameter.js
```

Comment:
Why pushing `ts-type-annotation-optional` when matching `...` now?

Maybe add a test case to clarify/justify that?

Reply:
It parses in AST explorer, so I allowed it. Test added.